### PR TITLE
IGVF-368-filter-dbinstanceidentifier-in-python

### PR DESCRIPTION
**Pull request by ottojolanki**

### CODE REVIEW

File: cdk/infrastructure/runtime/lambdas/snapshot/main.py [hunk 1]
```diff
@@ -47,6 +47,11 @@ def sort_results_by_create_time(results):
     )
 
 
+def filter_results_by_db_instance_identifier(results, db_instance_identifier):
+    filtered_results = [result for result in results if result['DBInstanceIdentifier'] == db_instance_identifier]
```
Comment (keenangraham) [inline cdk/infrastructure/runtime/lambdas/snapshot/main.py:51]: Everything will always have a `DBInstanceIdentifier`, right?

**Review by keenangraham — APPROVED** [review]
LGTM. We should try a couple different restore from snapshot scenarios on the demo.

File: cdk/infrastructure/runtime/lambdas/snapshot/main.py (hunk 1 above)
Comment (ottojolanki) [reply cdk/infrastructure/runtime/lambdas/snapshot/main.py:51]: Yes, that should be the case according to boto docs.

**Review by keenangraham — DISMISSED** [review]
I think this looks good.

Once the snaphot_share is deployed in production we should QA demo here that can:
1. Restore from latest dev database ID (like now).
2. Restore from latest prod database ID (new).
3. Restore from direct snapshot ID (local or shared).

**Review by keenangraham — APPROVED** [review]

**Pull request merged**